app/assets/javascripts/controllers/live_metrics/ad_hoc_metrics_controller.js

```diff
@@ -0,0 +1,125 @@
+/* global miqHttpInject */
+
+  ManageIQ.angular.app.controller('adHocMetricsController', ['$http', '$window', 'miqService',
+    'metricsUtilsFactory', 'metricsHttpFactory', 'metricsConfigFactory',
+    function($http, $window, miqService, metricsUtilsFactory, metricsHttpFactory, metricsConfigFactory) {
+
+    var dash = this;
+    var utils = metricsUtilsFactory(dash);
+    var httpUtils = metricsHttpFactory(dash, $http, utils, miqService);
+
+    // get the pathname and remove trailing / if exist
+    var pathname = $window.location.pathname.replace(/\/$/, '');
+
+    dash.providerId = '/' + (/^\/[^\/]+\/(\d+)$/.exec(pathname)[1]);
```

Comment:
Aah, I just created an issue (https://github.com/ManageIQ/manageiq-ui-classic/issues/560), looks like this could be relevant here.. Can you fix that `(\d+)` to say something like `([r\d]+)` please? So that compressed_ids are covered as well..